java/fury-core/src/main/java/org/apache/fury/serializer/NonexistentClassSerializers.java

```diff
@@ -241,13 +252,40 @@ public Object read(MemoryBuffer buffer) {
       return enumConstants[ordinal];
     }
   }
+    public static final class NonexistentEnumByNameClassSerializer extends Serializer {
+        private final Map<MetaStringBytes, NonexistentEnum> enumMap;
+        private final MetaStringResolver metaStringResolver;
+
+        public NonexistentEnumByNameClassSerializer(Fury fury) {
+            super(fury, NonexistentEnum.class);
+            metaStringResolver = new MetaStringResolver();
+            NonexistentEnum[] enumConstants = NonexistentEnum.class.getEnumConstants();
+            enumMap = Arrays.stream(enumConstants)
+                    .map(e -> {
+                        MetaStringBytes stringBytes =
+                                this.metaStringResolver.getOrCreateMetaStringBytes(GENERIC_ENCODER.encode(e.name(), MetaString.Encoding.UTF_8));
+                        return new AbstractMap.SimpleEntry<>(stringBytes, e);
+                    })
+                    .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+        }
 
+        @Override
+        public Enum read(MemoryBuffer buffer) {
+            MetaStringBytes metaStringBytes = metaStringResolver.readMetaStringBytes(buffer);
+            if (!enumMap.containsKey(metaStringBytes)) {
+                Optional<NonexistentEnum> first = enumMap.values().stream().findFirst();
```

Comment:
if `deserializeNonexistentEnumValueAsNull` is enabled, we should return null here, otherwise we should throw Exception

Reply:
Or instead of returning null, we could return `NonexistentEnum.UNKNOWN` instead